src/Plugin/Filter/EntityEmbedFilter.php

```diff
@@ -113,25 +107,33 @@ public function process($text, $langcode) {
 
             // Support the deprecated view-mode data attribute.
             if (isset($context['data-view-mode']) && !isset($context['data-entity-embed-display']) && !isset($context['data-entity-embed-settings'])) {
-              $context['data-entity-embed-settings']['view_mode'] = $context['data-view-mode'];
-              unset($context['data-view-mode']);
+              $context['data-entity-embed-display'] = 'default';
+              $context['data-entity-embed-settings'] = ['view_mode' => &$context['data-view-mode']];
             }
 
-            //$placeholder = $this->buildPlaceholder($entity, $result, $context);
-            //$this->setDomNodeContent($node, $placeholder);
+            // Merge in default attributes.
+            $context += array(
+              'data-entity-id' => $entity->id(),
+              'data-entity-embed-display' => 'default',
+              'data-entity-embed-settings' => array(),
+              'data-langcode' => $langcode,
+            );
 
             // Allow modules to alter the context.
             $this->moduleHandler()->alter('entity_embed_context', $context, $callback, $entity);
 
-            // Add cache tags and contexts.
-            if ($tags = $entity->getCacheTags()) {
-              $result->addCacheTags($tags);
-            }
-            if ($contexts = $entity->getCacheContexts()) {
-              $result->addCacheContexts($contexts);
-            }
+            $access = $entity->access('view', NULL, TRUE);
+            $accessMetaData = CacheableMetadata::createFromObject($access);
+            $entityMetadata = CacheableMetadata::createFromObject($entity);
 
-            if ($entity->access('view')) {
+            // @todo Swap these https://www.drupal.org/node/2516802 is fixed in core.
+            $result->addCacheContexts($accessMetaData->getCacheContexts());
+            $result->addCacheTags($accessMetaData->getCacheTags());
+            $result->addCacheContexts($entityMetadata->getCacheContexts());
+            $result->addCacheTags($entityMetadata->getCacheTags());
+            //$result = $result->merge($entityMetadata)->merge($accessMetadata);
```

Comment:
Let's leave it for now. This allows the module to work with D8 at least older than 24 hours at this point. :) We can swap this at the next beta or RC release.